### PR TITLE
[create-next-app] Update TypeScript & ESLint packages to be devDependencies

### PR DESCRIPTION
**Pull request by KoltonG**
Update the `create-next-app` package to install the following package as devDependencies:

**With `--typescipt`**
- `typescripts`
- `@types/node`
- `@types/react-dom`
- `@types/react`

**With ESLint**
- `eslint-config-next`
- `eslint`

Built and tested locally using the `/next.js/packages/create-next-app/dist/index.js --typescript` using the following hardware:'
- Mac v12.6 (21G115)

## Bug

- [X] Related issues linked using `fixes #number` - **No issue reported**
- [X] Integration tests added - **No tests for `create-next-app` test, however, local `pnpm test` have passed**
- [X] Errors have a helpful link attached, see `contributing.md`

## Feature

- [ ] Implements an existing feature request or RFC. Make sure the feature request has been accepted for implementation before opening a PR.
- [ ] Related issues linked using `fixes #number`
- [ ] Integration tests added
- [ ] Documentation added
- [ ] Telemetry added. In case of a feature if it's used or not.
- [ ] Errors have a helpful link attached, see `contributing.md`

## Documentation / Examples

- [ ] Make sure the linting passes by running `pnpm build && pnpm lint`
- [ ] The "examples guidelines" are followed from [our contributing doc](https://github.com/vercel/next.js/blob/canary/contributing/examples/adding-examples.md)

### CODE REVIEW

**Comment by KoltonG** [comment]
Removed the `pnpm-lock.yaml` file change as this was causing CI `pnpm install` failures.

File: packages/create-next-app/templates/index.ts [hunk 1]
```diff
@@ -94,6 +98,22 @@ export const installTemplate = async ({
...
+    for (const devDependency of devDependencies) {
+      console.log(`- ${chalk.cyan(devDependency)}`)
+    }
+    console.log()
+
+    await install(root, devDependencies, {
+      ...installFlags,
+      devDependencies: true,
+    })
+  }
```
Comment (KoltonG) [inline packages/create-next-app/templates/index.ts:116]: I just copy-pasted this section for the devDependencies addition, but I wasn't sure if you wanted to have all the install logs together vs in two places.

See the screenshot for an example.

CleanShot 2022-11-12 at 23 11 54@2x

**Comment by imranbarbhuiya** [comment]
check https://github.com/vercel/next.js/pull/42218 to know why it's installing all deps under normal deps

**Comment by KoltonG** [comment]
> check #42218 to know why it's installing all deps under normal deps

Thank you @imranbarbhuiya! Given #42218, I'll close this PR.

**Pull request closed without merging**